platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java

```diff
@@ -237,24 +244,32 @@ protected SecurityToken requestSecurityToken(Object authToken) {
      * @param securityToken The token being renewed.
      * @return security token (SAML assertion)
      */
-    protected SecurityToken renewSecurityToken(SecurityToken securityToken) {
+    protected SecurityToken renewSecurityToken(final SecurityToken securityToken) {
         SecurityToken token = null;
         String stsAddress = getAddress();
 
         try {
             LOGGER.debug("Renewing security token from STS at: {}.", stsAddress);
 
             if (securityToken != null) {
-                LOGGER.debug("Telling the STS to renew a security token on behalf of the auth token");
-                STSClient stsClient = configureStsClient();
-
-                stsClient.setWsdlLocation(stsAddress);
-                stsClient.setTokenType(getAssertionType());
-                stsClient.setKeyType(getKeyType());
-                stsClient.setKeySize(Integer.parseInt(getKeySize()));
-                stsClient.setAllowRenewing(true);
-                token = stsClient.renewSecurityToken(securityToken);
-                LOGGER.debug("Finished renewing security token.");
+                synchronized (securityToken.getToken()) {
+                    SecurityToken cachedToken = cache.getIfPresent(securityToken.getToken());
```

Comment:
But `expireAfterAccess` means if it was continually accessed (intervals less than the expiry time) it would continue to be returned after actual assertion expires even.